tests/engine/test_task_runner.py

```diff
@@ -978,13 +978,16 @@ def custom_validator(state, inputs, parameters):
         # have to have a state worth checking to trigger the validator
         with prefect.context(caches={"Task": [State()]}, checkpointing=False):
             task = Task(
-                cache_for=timedelta(seconds=10), cache_validator=custom_validator
+                cache_for=timedelta(seconds=10),
+                cache_validator=custom_validator,
+                task_run_name="TaskRun",
```

Comment:
related to my other comment, we'll want to have a test for both static and dynamic task run names, see examples of dynamic naming here https://docs.prefect.io/core/idioms/task-run-names.html